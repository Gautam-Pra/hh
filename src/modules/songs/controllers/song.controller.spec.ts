import { beforeAll, describe, expect, it } from 'vitest'
import type { z } from 'zod'
import type { SongModel } from '#modules/songs/models'
import { SongController } from '#modules/index'

describe('SongController', () => {
  let searchController: SongController

  beforeAll(() => {
    searchController = new SongController()
    searchController.initRoutes()
  })

  it('retrieve songs by link', async () => {
    const response = await searchController.controller.request(
      '/songs?link=https://www.jiosaavn.com/song/houdini/OgwhbhtDRwM'
    )

    const { data } = (await response.json()) as { data: z.infer<typeof SongModel>[] }

    expect(data[0]).toMatchSnapshot({
      playCount: expect.any(Number)
    })
  })

  it('retrieve song by ID', async () => {
    const response = await searchController.controller.request('/songs/3IoDK8qI')

    const { data } = (await response.json()) as { data: z.infer<typeof SongModel>[] }

    expect(data[0]).toMatchSnapshot({
      playCount: expect.any(Number)
    })
  })

  it('retrieve lyrics for a song', async () => {
    const response = await searchController.controller.request('/songs/ddQaeUpV/lyrics')

    const lyrics = await response.json()

    expect(lyrics).toMatchSnapshot()
  })

  // it('retrieve song suggestions', async () => {
  //   const response = await searchController.controller.request('/songs/yDeAS8Eh/suggestions?limit=5')
  //   const suggestions = await response.json()
  //   expect(suggestions).toMatchSnapshot()
  // })
})
