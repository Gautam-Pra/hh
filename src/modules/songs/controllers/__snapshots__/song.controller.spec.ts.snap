// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongController > retrieve lyrics for a song 1`] = `
{
  "data": {
    "copyright": "Lyrics powered by JioSaavn",
    "lyrics": "haq hai humko kuch aaj karle<br>dunia chahe aitraaz karle<br>bigde zyada sudhre huye kam<br>ab todenge har kayda hum<br><br>hosh nahi shaam se, kal milo kaam se<br>abhi toh hum dekh lo bilkul gaye<br>ke ghar ke saare raste bhul gaye<br>woh jo bhi hamein dekhe woh kehta rahe<br>ke ajj sher khul gaye<br><br>ghar ke raste bhul gaye<br>ab toh hum bilkul gaye<br>aaye aisi chorh mein<br>ke jo bhi hamein dekhe woh kehta rahe<br>ke ajj sher khul gaye<br><br>mujhse nashe mein galti ho jaani hai aaj toh<br>yaara sambhalna koi ho jaye naraaz jo<br>main aaj badmash ho raha hu, kal shareef tha,<br>bigda hisab dil ka pehle toh yeh thik tha,<br><br>hosh nahi shaam se, kal milo kaam se<br>abhi toh hum dekh lo bilkul gaye<br>ke ghar ke sare raste bhul gaye<br>woh jo bhi hamein dekhe woh kehta rahe<br>ke ajj sher khul gaye<br><br>ghar ke raste bhul gaye<br>ab toh hum bilkul gaye<br>aaye aisi chorh mein<br>ke jo bhi hamein dekhe woh kehta rahe<br>ke ajj sher khul gaye<br><br><br>",
    "snippet": "hosh nahi sham se, kal milo kaam se",
  },
  "success": true,
}
`;

exports[`SongController > retrieve song by ID 1`] = `
{
  "album": {
    "id": "23241654",
    "name": "Future Nostalgia",
    "url": "https://www.jiosaavn.com/album/future-nostalgia/ITIyo-GDr7A_",
  },
  "artists": {
    "all": [
      {
        "id": "573802",
        "image": [],
        "name": "Clarence Coffee Jr",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/clarence-coffee-jr-songs/2mX96rBxCYU_",
      },
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
      {
        "id": "702498",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/587/Gypsy-Girl-A-Cappella--English-2017-20191123100913-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/587/Gypsy-Girl-A-Cappella--English-2017-20191123100913-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/587/Gypsy-Girl-A-Cappella--English-2017-20191123100913-500x500.jpg",
          },
        ],
        "name": "Sarah Hudson",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/sarah-hudson-songs/pW4Y,NKAPPI_",
      },
      {
        "id": "599533",
        "image": [],
        "name": "Stephen Kozmeniuk",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/stephen-kozmeniuk-songs/lS1-2YuC5oc_",
      },
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "singer",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
    ],
    "featured": [],
    "primary": [
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "primary_artists",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
    ],
  },
  "copyright": "â„— 2020  Dua Lipa Limited under exclusive license to Warner Records UK, a division of Warner Music UK Limited",
  "downloadUrl": [
    {
      "quality": "12kbps",
      "url": "https://aac.saavncdn.com/665/7790c3b9097592113008eaf1031d6e57_12.mp4",
    },
    {
      "quality": "48kbps",
      "url": "https://aac.saavncdn.com/665/7790c3b9097592113008eaf1031d6e57_48.mp4",
    },
    {
      "quality": "96kbps",
      "url": "https://aac.saavncdn.com/665/7790c3b9097592113008eaf1031d6e57_96.mp4",
    },
    {
      "quality": "160kbps",
      "url": "https://aac.saavncdn.com/665/7790c3b9097592113008eaf1031d6e57_160.mp4",
    },
    {
      "quality": "320kbps",
      "url": "https://aac.saavncdn.com/665/7790c3b9097592113008eaf1031d6e57_320.mp4",
    },
  ],
  "duration": 203,
  "explicitContent": true,
  "hasLyrics": false,
  "id": "3IoDK8qI",
  "image": [
    {
      "quality": "50x50",
      "url": "https://c.saavncdn.com/665/Future-Nostalgia-English-2020-20231110075246-50x50.jpg",
    },
    {
      "quality": "150x150",
      "url": "https://c.saavncdn.com/665/Future-Nostalgia-English-2020-20231110075246-150x150.jpg",
    },
    {
      "quality": "500x500",
      "url": "https://c.saavncdn.com/665/Future-Nostalgia-English-2020-20231110075246-500x500.jpg",
    },
  ],
  "label": "Warner Records",
  "language": "english",
  "lyricsId": null,
  "name": "Levitating",
  "playCount": Any<Number>,
  "releaseDate": "2020-03-27",
  "type": "song",
  "url": "https://www.jiosaavn.com/song/levitating/QyEEdT8IRno",
  "year": 2020,
}
`;

exports[`SongController > retrieve songs by link 1`] = `
{
  "album": {
    "id": "49599287",
    "name": "Houdini",
    "url": "https://www.jiosaavn.com/album/houdini/YKL-I3to3Oc_",
  },
  "artists": {
    "all": [
      {
        "id": "776593",
        "image": [],
        "name": "Caroline Ailin",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/caroline-ailin-songs/ksTT3ixnrvU_",
      },
      {
        "id": "1264841",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/733/Super-Natural-Remixes-English-2016-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/733/Super-Natural-Remixes-English-2016-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/733/Super-Natural-Remixes-English-2016-500x500.jpg",
          },
        ],
        "name": "Danny L Harle",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/danny-l-harle-songs/8uc,o2eYVw8_",
      },
      {
        "id": "776593",
        "image": [],
        "name": "Caroline Ailin",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/caroline-ailin-songs/ksTT3ixnrvU_",
      },
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
      {
        "id": "1264841",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/733/Super-Natural-Remixes-English-2016-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/733/Super-Natural-Remixes-English-2016-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/733/Super-Natural-Remixes-English-2016-500x500.jpg",
          },
        ],
        "name": "Danny L Harle",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/danny-l-harle-songs/8uc,o2eYVw8_",
      },
      {
        "id": "657585",
        "image": [],
        "name": "Kevin Parker",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/kevin-parker-songs/WZROEDcg-u8_",
      },
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
      {
        "id": "1266339",
        "image": [],
        "name": "Tobias Jesso Jr.",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/tobias-jesso-jr.-songs/7n6h1Xoulxc_",
      },
      {
        "id": "657585",
        "image": [],
        "name": "Kevin Parker",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/kevin-parker-songs/WZROEDcg-u8_",
      },
      {
        "id": "1266339",
        "image": [],
        "name": "Tobias Jesso Jr.",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/tobias-jesso-jr.-songs/7n6h1Xoulxc_",
      },
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "singer",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
    ],
    "featured": [],
    "primary": [
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "primary_artists",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
    ],
  },
  "copyright": "Under exclusive license to Warner Records UK, a division of Warner Music UK Limited, â„— 2023 Radical22 Limited",
  "downloadUrl": [
    {
      "quality": "12kbps",
      "url": "https://aac.saavncdn.com/091/0d3cfd6080a23f13ace1c38c3296e18f_12.mp4",
    },
    {
      "quality": "48kbps",
      "url": "https://aac.saavncdn.com/091/0d3cfd6080a23f13ace1c38c3296e18f_48.mp4",
    },
    {
      "quality": "96kbps",
      "url": "https://aac.saavncdn.com/091/0d3cfd6080a23f13ace1c38c3296e18f_96.mp4",
    },
    {
      "quality": "160kbps",
      "url": "https://aac.saavncdn.com/091/0d3cfd6080a23f13ace1c38c3296e18f_160.mp4",
    },
    {
      "quality": "320kbps",
      "url": "https://aac.saavncdn.com/091/0d3cfd6080a23f13ace1c38c3296e18f_320.mp4",
    },
  ],
  "duration": 185,
  "explicitContent": false,
  "hasLyrics": false,
  "id": "JdJ_osp0",
  "image": [
    {
      "quality": "50x50",
      "url": "https://c.saavncdn.com/091/Houdini-English-2023-20231110015054-50x50.jpg",
    },
    {
      "quality": "150x150",
      "url": "https://c.saavncdn.com/091/Houdini-English-2023-20231110015054-150x150.jpg",
    },
    {
      "quality": "500x500",
      "url": "https://c.saavncdn.com/091/Houdini-English-2023-20231110015054-500x500.jpg",
    },
  ],
  "label": "Warner Records",
  "language": "english",
  "lyricsId": null,
  "name": "Houdini",
  "playCount": Any<Number>,
  "releaseDate": "2023-11-09",
  "type": "song",
  "url": "https://www.jiosaavn.com/song/houdini/OgwhbhtDRwM",
  "year": 2023,
}
`;
